src/util/shuffle.ts

```diff
@@ -0,0 +1,12 @@
+// function to randomly shuffle an array
+export function shuffle(array: any[]):any[] {
+    const arraycopy: any[]= [];
+    array.forEach((val: any) => arraycopy.push(Object.assign({}, val)));
+    for (var i = arraycopy.length - 1; i > 0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
```

Comment:
It would be better to use let and const here.

Reply:
Use let only if the variable is reassigned, const otherwise.

Reply:
Done, @harshkhandeparkar